docs/architecture/decisions/0001-organizing-code-under-features.md

```diff
@@ -0,0 +1,154 @@
+# 1. Organizing code under features
+
+Date: 2022-07-11
+
+## Status
+
+2022-07-11 proposed
+
+2022-08-10 accepted
+
+## Context
+
+Previously we have organized code under `main`, `renderer` and `common` top level directories. It means that given a feature which has code that needs to run in both environments, we had to split the feature in three different directories.
+
+**Example:** 
+
+```
+src/
+  behaviours/
+    application-update/
+      installing-update-using-tray.test.ts
+      ...
+      
+  main/
+    application-update
+      process-checking-of-updates.injectable.ts
+      ...
+      
+    tray/
+      tray-menu-item-injection-token.ts
+      check-for-updates-tray-menu-item.injectable.ts
+      ...
+     
+  common/
+    application-update
+      update-channels.ts
+      update-is-being-downloaded.injectable.ts
+      ...
+      
+    front-end-routes/
+      preferences/
+        app/
+          app-preferences-route.injectable.ts
+          navigate-to-app-preferences.injectable.ts
+
+  renderer/
+    top-bar/
+      top-bar.tsx (UpdateButton)
+    
+    preferences/
+      application.tsx (UpdateChannel)
+```
+
+## Decision
+
+Start organizing code under features, meaning that all code that feature needs to exist in a single directory. Features might have child features which have dependency to the parent. We want to limit the coupling of features, which means that feature is not allowed to touch any other feature without creating dependency to it. We want to have visibility for dependency structures of features.
+
+1. Code that belongs to the feature lives in the feature directory. If I want to remove the feature, I can just delete the directory.
+2. Code that is shared between features lives in the closest parent feature. E.g. child features are using code from parent features.
+3. Parent features should never depend on child features. If the dependency is required, the child feature is the one that depends on the parent feature.
```

Comment:
This point also seems too prescriptive and backwards since it feels strange to go to the leaves of a feature directory to see the "actual implementation".